### PR TITLE
Feature/page recharge

### PR DESCRIPTION
**Pull request by juanca172**
Simulator Screen Recording - Iphone 15 - 2023-10-05 at 16 14 49

### CODE REVIEW

File: RickAndMorty/Models/RMCharacter.swift [hunk 1]
```diff
@@ -34,7 +34,7 @@ struct RMCharacter: Decodable  {
     enum Gender: String, Decodable {
         case male = "Male"
         case female = "Female"
-        case genderless
+        case Genderless
```
Comment (SSPF20) [inline RickAndMorty/Models/RMCharacter.swift]: for us all cases must be lowercased

File: RickAndMorty/Views/RMCharacterList/RMCharacterHorizontalCell.swift [hunk 2]
```diff
@@ -18,6 +18,7 @@ final class RMCharacterHorizontalCell: UICollectionViewCell, RMItemCell {
     private var locationLabel = UILabel()
     
     
+
```
Comment (SSPF20) [inline RickAndMorty/Views/RMCharacterList/RMCharacterHorizontalCell.swift]: remove extra line here

File: RickAndMorty/Models/RMCharacter.swift (hunk 1 above)
Comment (juanca172) [reply RickAndMorty/Models/RMCharacter.swift]: changed done

File: RickAndMorty/Views/RMCharacterList/RMCharacterHorizontalCell.swift (hunk 2 above)
Comment (juanca172) [reply RickAndMorty/Views/RMCharacterList/RMCharacterHorizontalCell.swift]: done

File: RickAndMorty/Models/RMCharacter.swift [hunk 3]
```diff
@@ -34,7 +34,7 @@ struct RMCharacter: Decodable  {
     enum Gender: String, Decodable {
         case male = "Male"
         case female = "Female"
-        case genderless
+        case genderless = "Genderless"
```
Comment (hgp691) [inline RickAndMorty/Models/RMCharacter.swift]: nice!

File: RickAndMorty/Extensions/String+Ext.swift [hunk 4]
```diff
@@ -18,6 +18,14 @@ extension String {
         return nil
     }
     
+    var pageNumber: Int {
```
Comment (hgp691) [inline RickAndMorty/Extensions/String+Ext.swift]: Should be Optional `Int?`

File: RickAndMorty/Views/Footer/FooterView.swift [hunk 5]
```diff
@@ -0,0 +1,44 @@
+//
+//  FooterViewCollectionReusableView.swift
+//  RickAndMorty
+//
+//  Created by Juan Camilo Fonseca Gomez on 2/08/23.
+//
+
+import UIKit
+
+class FooterView: UICollectionReusableView {
```
Comment (hgp691) [inline RickAndMorty/Views/Footer/FooterView.swift]: should be `final`

File: RickAndMorty/Views/Footer/FooterView.swift [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    
+    static let identifier = "FooterView"
+    
+    private let spinner = UIActivityIndicatorView()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        layout()
```
Comment (hgp691) [inline RickAndMorty/Views/Footer/FooterView.swift]: naming

File: RickAndMorty/Views/RMListView/RMListViewController.swift [hunk 7]
```diff
@@ -99,17 +105,37 @@ final class RMListViewController<A: Decodable, B: Configuration>: UIViewControll
     }
     
     
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
```
Comment (hgp691) [inline RickAndMorty/Views/RMListView/RMListViewController.swift:109]: check if should be in extension

File: RickAndMorty/Views/RMListView/RMListViewModel.swift [hunk 8]
```diff
@@ -51,24 +53,41 @@ final class RMListViewModel<A: Decodable, B: Configuration> {
...
-        Task {
-            do {
-                let response = try await dataProvider.getPage(page: page)
-                elements.append(response.results)
-                currentInfo = response.info
-                
-            } catch {
-                print("LoadPage error \(String(describing: error))")
-            }
+        if isPaginando == false {
```
Comment (hgp691) [inline RickAndMorty/Views/RMListView/RMListViewModel.swift]: ?

File: RickAndMorty/Views/RMListView/RMListViewModel.swift [hunk 9]
```diff
@@ -51,24 +53,41 @@ final class RMListViewModel<A: Decodable, B: Configuration> {
...
+            Task {
+                    do {
```
Comment (hgp691) [inline RickAndMorty/Views/RMListView/RMListViewModel.swift:88]: ident ?

File: RickAndMorty/Views/RMListView/RMListViewController.swift (hunk 7 above)
Comment (juanca172) [reply RickAndMorty/Views/RMListView/RMListViewController.swift:109]: I can't attach a method in extensions

File: RickAndMorty/Extensions/String+Ext.swift [hunk 10]
```diff
@@ -18,6 +18,14 @@ extension String {
         }
         return nil
     }
+    
+    var pageNumber: Int? {
```
Comment (hgp691) [inline RickAndMorty/Extensions/String+Ext.swift]: This changed with the Property Wrapper approach

**Review by hgp691 — APPROVED** [review]

**Pull request merged**